File: ProductsSolution/Entities/SalePoint.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Entities
{
    public class SalePoint : BaseEntity
    {
        public string Description { get; set; }
        public string Address { get; set; }
      
        public int CountryId { get; set; }
        
        [ForeignKey("CountryId")]
        public Country Country { get; set; }
    }
}

File: shopping/src/app/Model/SalePoint.ts
export class SalePoint {
    id: number;
    description: string;
    address: string;
    countryId: number;
    country: string;
    countryCategory: string;

    constructor(id: number, description: string, address: string, country: string,
      countryId: number,
      countryCategory: string,
    ) {
        this.id = id;
        this.description = description;
        this.address = address;
        this.country = country;
        this.countryId = countryId;
        this.countryCategory = countryCategory;
    }
}
